x-pack/platform/plugins/shared/alerting/server/application/rule/transforms/transform_raw_artifacts_to_domain_artifacts.test.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectReference } from '@kbn/core/server';
+import type { RawRule } from '../../../types';
+import { transformRawArtifactsToDomainArtifacts } from './transform_raw_artifacts_to_domain_artifacts';
+
+describe('transformRawArtifactsToDomainArtifacts', () => {
+  it('should return empty artifacts if rawArtifacts is undefined', () => {
+    const result = transformRawArtifactsToDomainArtifacts('1', undefined, []);
+    expect(result).toEqual({ dashboards: [] });
+  });
+
+  it('should return empty object if rawArtifacts is empty', () => {
+    const result = transformRawArtifactsToDomainArtifacts('1', {}, []);
+    expect(result).toEqual({});
+  });
+
+  it('should return artifacts with injected references', () => {
+    const rawArtifacts: RawRule['artifacts'] = {
+      dashboards: [
+        {
+          refId: 'dashboard-1',
+        },
+        {
+          refId: 'dashboard-2',
+        },
+      ],
+    };
+    const references: SavedObjectReference[] = [
+      {
+        id: 'dashboard_0',
+        name: 'dashboard-1',
+        type: 'dashboard',
+      },
+      {
+        id: 'dashboard_1',
+        name: 'dashboard-2',
+        type: 'dashboard',
+      },
+    ];
+    const result = transformRawArtifactsToDomainArtifacts('1', rawArtifacts, references);
+    expect(result).toEqual({
+      dashboards: [
+        {
+          id: 'dashboard_0',
+        },
+        {
+          id: 'dashboard_1',
+        },
+      ],
+    });
+  });
+
+  it('should return artifacts with empty dashboards array if no dashboards in rawArtifacts', () => {
```

Comment:
How is this test different than the second test?

Reply:
They're identical, but I think it's actually a bug. The return should always have an empty dashboards array nested in the object shouldn't it? I will update the expect of this one, and I think we should not be expecting an empty object ever, should we?

Reply:
See c5ba626b932beb446f0e4220e3caa4aea1971e2e